### ISSUE TITLE
Return the HttpStatusCode on each request

### ISSUE THREAD

**Issue by claudiuolteanu** [skrusty/AsterNET.ARI#14]
Hi there,

It would be useful if the framework will return the HTTP status code after you execute a HTTP request.

For example **EventsActions::UserEvent** method should return the **IRestCommandResult** from the execute method (or raise an exception).

Cheers,
Claudiu

**Comment by skrusty** [comment]
In most cases it does at least return an exception, and agreed, as it's a customer AriException there's no reason why we shouldn't include the status code there.

When the return type is say a Channel we'd have to wrap that return type in something to give the status code (or have an out param, but I don't like the idea of that).

Where there's no return type, then I don't see why we couldn't return the status code!

Your specific example however is interesting. AS that code is generated from the Swagger JSON i can only assume there's nothing in the JSON defining any http response codes. I will need to look into that and let you know (they're in the repo too by the way if you want to check yourself).

**Comment by skrusty** [comment]
JSON seems to say there are httpresponse codes!

```json
"errorResponses": [
						{
							"code": 404,
							"reason": "Application does not exist."
						},
						{
							"code": 422,
							"reason": "Event source not found."
						},
						{
							"code": 400,
							"reason": "Invalid even tsource URI or userevent data."
						}
					]
```
So not quite sure why it's not in the code, will need to run the code gen and find out why they're not showing up.

**Comment by skrusty** [comment]
The issue is the events file is swagger v1.2 not 1.1. We will need to make very specific changes to the codegen.

**Comment by skrusty** [comment]
Fixed in commit 210b7f967b7dcf1573dbc3c72b5cfba4ff7795e0
